airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/LauncherWorker.java

```diff
@@ -136,28 +151,49 @@ public OUTPUT run(INPUT input, Path jobRoot) throws WorkerException {
   }
 
   /**
-   * If the sync workflow has advanced to the next attempt, we don't want to leave a zombie of the
-   * older job running (if it exists). In order to ensure a consistent state, we should kill the older
-   * versions.
+   * It is imperative that we do not run multiple replications, normalizations, syncs, etc. at the
+   * same time. Our best bet is to kill everything that is labelled with the connection id and wait
+   * until no more pods exist with that connection id.
    */
-  private void killLowerAttemptIdsIfPresent(final String podNameAndJobPrefix, final long currentAttempt) {
-    for (long previousAttempt = currentAttempt - 1; previousAttempt >= 0; previousAttempt--) {
-      final var podName = podNameAndJobPrefix + previousAttempt;
-      final var kubePodInfo = new KubePodInfo(containerOrchestratorConfig.namespace(), podName);
-      final var oldProcess = new AsyncOrchestratorPodProcess(
-          kubePodInfo,
-          containerOrchestratorConfig.documentStoreClient(),
-          containerOrchestratorConfig.kubernetesClient());
+  private void killRunningPodsForConnection() {
+    final var client = containerOrchestratorConfig.kubernetesClient();
 
-      try {
-        oldProcess.destroy();
-        log.info("Found and destroyed a previous attempt: " + previousAttempt);
-      } catch (Exception e) {
-        log.warn("Wasn't able to find and destroy a previous attempt: " + previousAttempt);
+    // delete all pods with the connection id label
+    final var runningPods = getNonTerminalPodsWithLabels();
+
+    if (!runningPods.isEmpty()) {
+      log.warn("There are currently running pods for the connection: " + getPodNames(runningPods).toString());
+
+      runningPods.stream().forEach(kubePod -> client.resource(kubePod).withPropagationPolicy(DeletionPropagation.FOREGROUND).delete());
```

Comment:
Should we use parallel stream If we end up having hundred of pods the waiting time might become an issue.

Reply:
Done. It never should get in the state where there are more than a single other step (like an old replication) that's running when a new replication is trying to start, but this will definitely help if that expectation isn't met.